File: app/src/main/java/com/playgilround/schedule/client/tutorial/TutorialContract.java
package com.playgilround.schedule.client.tutorial;

import com.playgilround.schedule.client.base.BasePresenter;
import com.playgilround.schedule.client.base.BaseView;

/**
 * 19-02-20
 * MVP Pattern
 * Contract -> View, Presenter에 대한 interface
 */
public interface TutorialContract {

    interface View extends BaseView<Presenter> {

        void skipTutorial();

    }

    interface Presenter extends BasePresenter {

        void checkLogin();

    }
}

File: app/src/main/java/com/playgilround/schedule/client/tutorial/TutorialContract.kt
package com.playgilround.schedule.client.tutorial

import com.playgilround.schedule.client.base.BasePresenter
import com.playgilround.schedule.client.base.BaseView

/**
 * 19-08-07
 */

interface TutorialContract {
    
    interface View: BaseView<Presenter> {
        fun skipTutorial()
    }

    interface Presenter: BasePresenter {
        fun checkLogin()
    }
}